setup-app.py

```diff
@@ -1,19 +1,72 @@
+import requests
 import yaml
-from subprocess import call
+from subprocess import call, Popen, PIPE
+import sys
+import time
+
+
+def domain_has_ssl(domain, print_info=False):
+    """domain_has_ssl uses the two most-reliable ways to check for an SSL on
+    a domain name within bluemix. It calls the Bluemix CLI to ask for
+    certificate details, and it attempts to connect to that domain with HTTPS.
+    If either succeeds, it returns True. Otherwise, it returns false. Note
+    that it is possible to get false negatives, but not false positives.
+    The print_info parameter can be used to dump the certificate information
+    from Bluemix to stdout.
+    """
+    pipe = Popen("bx security cert %s" % domain,
+                 stdout=PIPE, shell=True)
+    output = pipe.stdout.read()
+    cert_exists = "OK" in output
+    if print_info and cert_exists:
+        print(output)
+    return cert_exists or check_ssl(domain)
+
+
+def get_cert(appname, domain, certname):
+    """get_cert wraps the `cf files` command to retrive only the literal file
+    contents of the certificate that was requested, without the status code at
+    the beginning. It then writes the certificate to a file in the current
+    working directory with the same name that the certificate had on the
+    server.
+    """
+    command = "cf files %s app/conf/live/%s/%s" % (appname, domain, certname)
+    print("Running: %s" % command)
+    pipe = Popen(command, shell=True, stdout=PIPE)
+    output = pipe.stdout.readlines()
+    cert = ''.join(output[3:-1])  # Strip leading and trailing characters
+    with open(certname, 'w') as outfile:
+        print("Writing cert to %s" % certname)
+        outfile.write(cert)
+
+
+def check_ssl(ssl_domain):
+    """check_ssl makes an HTTPS request to a given domain name and
+    returns a boolean for whether the SSL on the domain is present
+    and valid.
+    """
+    try:
+        target = "https://%s" % ssl_domain
+        print("Making GET request to %s" % target)
+        requests.get(target)
+        return True
+    except requests.exceptions.SSLError as err:
+        print(err)
+        return False
 
 with open('domains.yml') as data_file:
```

Comment:
It took some effort to identify this as the 'start' of the script, and so I looked up the pattern:
http://stackoverflow.com/questions/3754240/declare-function-at-end-of-file-in-python

I kinda like the accepted answer, although the commenters claim that the only thing 'idiomatic' about that is using ```if __name__ == '__main__'``` 

Maybe at least we should leave a short comment to help the reader?